__tests__/package-resolver.js

```diff
@@ -53,6 +53,7 @@ function addTest(pattern, registry = 'npm', init: ?(cacheFolder: string) => Prom
 // addTest('https://github.com/npm-ml/re'); // git url with no .git
 // addTest('git+https://github.com/npm-ml/ocaml.git#npm-4.02.3'); // git+hash
 // addTest('https://github.com/npm-ml/ocaml.git#npm-4.02.3'); // hash
+addTest('https://github.com/mozilla/webextension-polyfill#99faff94ffbf03094361875a10e1692acc4b8662'); // commit in pull request
```

Comment:
https://github.com/yarnpkg/e2e-test-repo may be a more appropriate location to create a pull request (and leave it open) for this test